code/game/machinery/camera/camera.dm

```diff
@@ -294,6 +295,18 @@
 	else
 		icon_state = initial(icon_state)
 
+	pixel_x = 0
+	pixel_y = 0
+
+	var/turf/T = get_step(get_turf(src), turn(src.dir, 180))
+	if(istype(T, /turf/simulated/wall))
+		if(dir == SOUTH)
+			pixel_y = 21
+		else if(dir == WEST)
+			pixel_x = 10
+		else if(dir == EAST)
+			pixel_x = -10
```

Comment:
this should be in set_dir and/or triggered by that + initialize as with above, not update_icon